gtsfm/frontend/verifier/gric_verifier.py

```diff
@@ -0,0 +1,191 @@
+    CALIBRATED = 2
+    # Fundamental matrix.
+    UNCALIBRATED = 3
+    # Homography, planar scene with baseline.
+    PLANAR = 4
+    # Homography, pure rotation without baseline.
+    PANORAMIC = 5
+    # Homography, planar or panoramic.
+    PLANAR_OR_PANORAMIC = 6
+    # Watermark, pure 2D translation in image borders.
+    WATERMARK = 7
+    # Multi-model configuration, i.e. the inlier matches result from multiple
+    # individual, non-degenerate configurations.
+    MULTIPLE = 8
+
+
+class GricVerifier(VerifierBase):
+    def __init__(
+        self,
+        use_intrinsics_in_verification: bool,
+        estimation_threshold_px: float,
+    ) -> None:
+        """Initializes the verifier.
+
+        (See https://github.com/mihaidusmanu/pycolmap/blob/master/essential_matrix.cc#L98)
+
+        Args:
+            use_intrinsics_in_verification: Flag to perform keypoint normalization and compute the essential matrix
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known.
+            estimation_threshold_px: maximum distance (in pixels) to consider a match an inlier, under squared
+                Sampson distance.
+        """
+        self._use_intrinsics_in_verification = use_intrinsics_in_verification
+        self._estimation_threshold_px = estimation_threshold_px
+        self._min_matches = (
+            verifier_base.NUM_MATCHES_REQ_E_MATRIX
+            if self._use_intrinsics_in_verification
+            else verifier_base.NUM_MATCHES_REQ_F_MATRIX
+        )
+
+        # for failure, i2Ri1 = None, and i2Ui1 = None, and no verified correspondences, and inlier_ratio_est_model = 0
+        self._failure_result = (None, None, np.array([], dtype=np.uint64), 0.0)
+
+    def __estimate_two_view_geometry(
+        self,
+        uv_i1: np.ndarray,
+        uv_i2: np.ndarray,
+        camera_intrinsics_i1: Cal3Bundler,
+        camera_intrinsics_i2: Cal3Bundler,
+    ) -> Dict[str, Any]:
+        """Use the pycolmap Pybind wrapper to estimate an Essential matrix using LORANSAC.
+
+        Args:
+            uv_i1: array of shape (N3,2) representing coordinates of 2d points in image 1.
+            uv_i2: array of shape (N3,2) representing corresponding coordinates of 2d points in image 2.
+            camera_intrinsics_i1: intrinsics for image i1.
+            camera_intrinsics_i2: intrinsics for image i2.
+
+        Returns:
+            dictionary containing result status code, estimated relative pose (R,t), and inlier mask.
+        """
+
+        def get_pycolmap_camera(camera_intrinsics: Cal3Bundler) -> pycolmap.Camera:
+            """Convert Cal3Bundler intrinsics to a pycolmap-compatible format (a dictionary).
+
+            See https://colmap.github.io/cameras.html#camera-models for info about the COLMAP camera models.
+            Both SIMPLE_PINHOLE and SIMPLE_RADIAL use 1 focal length.
+            """
+            focal_length = camera_intrinsics.fx()
```

Comment:
Take mean of fx and fy?

Reply:
Cal3Bundler only has a single common focal length (https://github.com/borglab/gtsam/blob/develop/gtsam/geometry/Cal3Bundler.h#L61), so we don't know fy